hummedia/resources.py

```diff
@@ -1,1041 +1,1076 @@
-            
-        pid = bundle['@graph']['pid']
-
-        vid=assets.find_one(bundle["@graph"]["dc:relation"])
-        required = True if pid in vid["@graph"].get("ma:hasPolicy") else False
-
-        if required and atts['role'] == 'faculty':
-            return True
-
-        return can_write_to_vid(vid, name)
-        
-    def set_disallowed_atts(self):
-        from auth import get_profile
-        atts=get_profile()
-        if not atts['superuser']:
-            self.disallowed_atts.append("dc:creator")
-    
-    def set_attrs(self):
-        import traceback
-        from flask import send_file
-        if "client" in self.request.args:
-            client_name = self.request.args.get("client")
-            c=clients.lookup[client_name]()
-            packet=c.deserialize(self.request)
-        else:
-            packet=self.request.json
-        for (k,v) in packet.items():
-            if k=="dc:title":
-                self.bundle["@graph"]["dc:title"]=unicode(v)
-            elif k=="vcp:playSettings" and v:
-                if self.bundle["@graph"]["vcp:playSettings"] is None:
-                    self.bundle["@graph"]["vcp:playSettings"] = {}
-                for (i,j) in v.items():
-                    if i=="vcp:frameRate":
-                        self.bundle["@graph"]["vcp:playSettings"][i]=float(j)
-                    elif i=="vcp:videoCrop":
-                        self.bundle["@graph"]["vcp:playSettings"][i]=j
-                    else:
-                        self.bundle["@graph"]["vcp:playSettings"][i]=int(j)
-            elif k=="vcp:commands":
-                self.bundle["@graph"]["vcp:commands"]=[]
-                for i in v:
-                    self.bundle["@graph"]["vcp:commands"].append(i)
-            else:
-                self.bundle["@graph"][k]=v
-        if self.request.method == "POST":
-            if "collection" in self.request.args:
-		assets.update({"_id":self.bundle["@graph"]["dc:relation"],"@graph.ma:isMemberOf.@id":self.request.args["collection"]},{"$set":{"@graph.ma:isMemberOf.$.restrictor":self.bundle["_id"]}})
-            else:
-		assets.update({"_id":self.bundle["@graph"]["dc:relation"]},{"$addToSet":{"@graph.ma:hasPolicy":self.bundle["_id"]}})
-	return ({"resp":200})
+from datetime import datetime, timedelta
+from os.path import splitext
+from models import connection
+from flask import request, Response, jsonify
+from helpers import Resource, mongo_jsonify, parse_npt, plain_resp, resolve_type, uri_pattern, bundle_400, bundle_404, action_401, is_enrolled, getYtThumbs
+from mongokit import cursor
+from bson import ObjectId
+from urlparse import urlparse, parse_qs
+import clients, config, json, re, hashlib, time
+from hummedia import app
+from os import system, chmod, chdir, getcwd, listdir, path, remove
+import vtt
+
+db=connection[config.MONGODB_DB]
+ags=db.assetgroups
+assets=db.assets
+annotations=db.annotations
+users=db.users
+
+class UserProfile(Resource):
```

Comment:
IMO they should all be switched in one fell swoop on a separate commit. Otherwise working through the revision history could become nightmarish (if a small feature/bug change altered every line in the file and we needed to revert it, for example).